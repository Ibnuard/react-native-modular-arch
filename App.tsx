import 'react-native-gesture-handler';
import * as React from 'react';
import {LogBox} from 'react-native';
import MainApp from './src';

LogBox.ignoreAllLogs();

const App = () => {
  return <MainApp />;
};

export default App;
