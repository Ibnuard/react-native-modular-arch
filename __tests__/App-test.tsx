import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import App from '../App';

const sleep = (ms: number) =>
  new Promise(resolve => setTimeout(() => resolve(true), ms));

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

it('renders correctly', async () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
  await sleep(10);
});
