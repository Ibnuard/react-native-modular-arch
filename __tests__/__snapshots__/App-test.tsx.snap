// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="SignIn"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RNCSafeAreaView
            edges={
              {
                "bottom": "additive",
                "left": "additive",
                "right": "additive",
                "top": "additive",
              }
            }
            style={
              {
                "flex": 1,
                "padding": 47.368421052631575,
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#A3A3A3",
                    "borderRadius": 8,
                    "borderWidth": 0.5,
                    "paddingHorizontal": 19.736842105263158,
                  },
                  undefined,
                ]
              }
            >
              <TextInput
                placeholder="Masukan Email"
                placeholderTextColor="#A3A3A3"
                style={
                  {
                    "color": "#090909",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 24,
                    "fontWeight": "normal",
                  }
                }
              />
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffa418",
                  "borderRadius": 8,
                  "justifyContent": "center",
                  "marginTop": 23.684210526315788,
                  "opacity": 1,
                  "paddingVertical": 23.684210526315788,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 24,
                    "fontWeight": "normal",
                  }
                }
              >
                Masuk
              </Text>
            </View>
          </RNCSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
